samcli/lib/providers/sam_layer_provider.py

```diff
@@ -105,8 +104,8 @@ def _convert_lambda_layer_resource(
 
         return LayerVersion(
             layer_logical_id,
-            codeuri,
+            codeuri and SamLocalStackProvider.normalize_resource_path(stack, codeuri),
```

Comment:
Why is the `codeuri and ...` needed?

Reply:
`codeuri` could be `None`, so I need to check it first.

Reply:
Would it make sense to check that in `normalize_resource_path` instead?

Reply:
I made it consistent with built in path functions, such as `os.path.join(a, b)` or `Path(a, b)`, functions with "path" in their name do not accept `None` as an argument.